sdk-core-protos/src/lib.rs

```diff
@@ -129,6 +122,26 @@ pub mod coresdk {
             pub fn failed(&self) -> bool {
                 matches!(self.status, Some(activity_resolution::Status::Failed(_)))
             }
+
+            pub fn cancelled(&self) -> bool {
+                matches!(self.status, Some(activity_resolution::Status::Cancelled(_)))
+            }
+        }
+
+        impl Cancellation {
+            pub fn from_details(payload: Option<Payload>) -> Self {
```

Comment:
Pedantic, but maybe "activity_cancelled_with_details" or something if `"Activity cancelled"` is hardcoded in this utility.

Reply:
This struct can only go inside of an activity result, so that's implied by the types - the `from_` prefix is idiomatic rust here when consuming one value to produce another.

Reply:
Ah! I was using that contextual info at the top of the GH bar and thought this was in `coresdk`, I didn't see it was in `activity_result` mod.